File: moban_java/src/main/java/org/jdk/project/dto/knowledge/KnowledgeUpsertRequest.java
package org.jdk.project.dto.knowledge;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class KnowledgeUpsertRequest {
  private String title;
  private String content;
  private String source;
  private String knowledgeType;
  private String processStatus;
  private Boolean enabled;
  private String vectorRef;
  private Long createdBy;
  private String remark;
}

File: backend/src/main/kotlin/org/jdk/project/dto/knowledge/KnowledgeUpsertRequest.kt
package org.jdk.project.dto.knowledge

data class KnowledgeUpsertRequest(
    var title: String? = null,
    var content: String? = null,
    var source: String? = null,
    var knowledgeType: String? = null,
    var processStatus: String? = null,
    var enabled: Boolean? = null,
    var vectorRef: String? = null,
    var createdBy: Long? = null,
    var remark: String? = null,
)
